### ISSUE TITLE
Beemod causes "Failure To Compile", help please!!!

### ISSUE THREAD

**Issue by lcyLand** [BEEmod/BEE2.4#943]
So, remember that one issue I opened because the bee mod wouldn’t compile anything? Well, I kind of found out the problem, but I have a new problem at the same time. The original problem was with VBSP, which at the time I thought the problem was with the mod. But no, it was a problem with the items in the chamber. I tried running the map through hammer, and it also failed to run. The error was “Failed to open instances\BEE2\cube_coloriser.vmf” or something like that. So I reopened the level, deleted the cube coloriser things, but a new error appeared. Can you guess what it is? If you said with VBSP, well, it wasn’t. The problem is now VRAD. It will build geometry, compute visibility, then by the time calculating lighting is almost done, it fails. I’ll have to copy-paste the error later. So, this is a new problem that I do need help with.

**Comment by lcyLand** [comment]
Can anybody help?

**Comment by ZombiePwner1** [comment]
@lcyLand Open up the console in portal 2 and see what the error message says.

**Comment by lcyLand** [comment]
Ok. I forgot to do it earlier, and I just woke up now.

**Comment by lcyLand** [comment]
VRAD failed.

[INFO] vrad.main(): BEE2 VRAD hook started!
[INFO] vrad.main(): Map path is C:\Program Files (x86)\Steam\steamapps\common\Portal 2\sdk_content\maps\preview.bsp
[INFO] vrad.load_config(): Loading Settings...
[INFO] vrad.load_config(): Config Loaded!
[INFO] vrad.main(): Final status: is_peti=True, edit_args=True
[INFO] vrad.main(): Reading BSP
[INFO] vrad.main(): Done!
[INFO] vrad.gen_sound_manifest(): Written new soundscripts_manifest..
[INFO] vrad.pack_content(): Files to pack:
[INFO] vrad.pack_content():  # "materials/bee2/fizz/fizz_side.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/fizz_sides/side_color_06141a.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/fizz_sides/side_color_ff0000.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/glass_player_clip.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/base.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_arrow.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_arrow.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_exit.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_exit.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape1.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape1.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape2.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape2.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape4.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape4.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape5.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape5.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape_slash.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/josepezdj/signage_wasted/jose_wasted_signage_shape_slash.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/cust_fizz/effect.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/cust_fizz/fizzler_dirty_danger.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/cust_fizz/fizzler_dirty_danger.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/bounce.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp_bounce.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp_on.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp_rusty.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp_rusty_on.vtf"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/comp_speed.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/on_bounce.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/on_speed.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/metal_box_rusty/speed.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/reflecto_cube_dirty_bounce.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/reflecto_cube_dirty_speed.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_ingame/triple_wallpipe_nocull.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_overgrown/catwalk_rusty.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_overgrown/triple_wallpipe_nocull_rusty.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_overgrown/triple_wallpipe_rusty.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/models/props_overgrown/truss_rusty.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/broken_indicator.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/broken_indicator_corner.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/glass.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/screens/elevator_video_overlay1.vmt" as "materials/bee2/overgrown/lightboard_overlay.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/tile/white_floor_tile005a.vmt"
[INFO] vrad.pack_content():  # "materials/bee2/overgrown/tile/white_floor_tile005a.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_a_overgrown.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_a_overgrown.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_a_overgrown_off.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_a_overgrown_off.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_aoff.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_aoff.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_b_overgrown.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_b_overgrown.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_b_overgrown_off.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_boff.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_c_overgrown.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_c_overgrown.vtf"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_c_overgrown_off.vmt"
[INFO] vrad.pack_content():  # "materials/models/elevator/elevator_main_coff.vmt"
[INFO] vrad.pack_content():  # "models/bee2/catwalk/overgrown/hanging_stair_128.mdl"
[INFO] vrad.pack_content():  # "models/bee2/catwalk/overgrown/hanging_walkway_32b.mdl"
[INFO] vrad.pack_content():  # "models/bee2/catwalk/overgrown/hanging_walkway_end_cap.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/cracked_glass/64x64_a.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/cracked_glass/64x64_b.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/cracked_glass/64x64_c.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/cust_fizz.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/cust_fizz_static.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/metal_box_rusty.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/pgun_pedestal.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_ingame/reflection_cube_rusty.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_overgrown/angled_pan.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_overgrown/truss_1024.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_overgrown/truss_512.mdl"
[INFO] vrad.pack_content():  # "models/bee2/props_overgrown/wrecked_dropper_128.mdl"
[INFO] vrad.pack_content():  # "models/bee2/triple_wallpipe/corner_b.mdl"
[INFO] vrad.pack_content():  # "models/bee2/triple_wallpipe/corner_c.mdl"
[INFO] vrad.pack_content():  # "models/bee2/triple_wallpipe/straight_a.mdl"
[INFO] vrad.pack_content():  # "models/bee2/triple_wallpipe/straight_c.mdl"
[INFO] vrad.pack_content():  # "models/elevator/elevator_b_skins.mdl"
[INFO] vrad.pack_content():  # "scripts/vscripts/bee2/change_models.nut"
[INFO] vrad.pack_content():  # "scripts/vscripts/bee2/portal_man.nut"
[INFO] vrad.pack_content():  # "scripts/vscripts/bee2/video_splitter_specific.nut"
[INFO] vrad.pack_content():  # "sound/bee2/p1/charge_pgun.wav"
[INFO] vrad.pack_content():  # "sound/music/sp_a1_intro6.wav"
[INFO] vrad.pack_content():  # "sound/music/sp_a3_bomb_flings_z2.wav"
[INFO] vrad.pack_content():  # "sound/music/sp_a3_bomb_flings_z2b.wav"
[INFO] vrad.pack_content():  # "sound/music/sp_a3_bomb_flings_z2c.wav"
[INFO] vrad.pack_content():  # "sound/music/sp_a4_tb_intro_tbin.wav"
[INFO] vrad.pack_content(): Potential additional files:
[INFO] vrad.pack_content(): Injected files:
[INFO] vrad.pack_content():  # "scripts/game_sounds_manifest.txt"
[INFO] vrad.pack_content():  # "scripts/BEE2_generated_music.txt"
[INFO] vrad.pack_content():  # "scripts/vscripts/bee2/auto_run.nut"
[INFO] vrad.pack_content():  # "scripts/vscripts/BEE2/cube_setmodel.nut"
[INFO] vrad.pack_content(): Packing Files!
[WARNING] vrad.pack_file(): "bee2/sound/music/sp_a4_tb_intro_tbin.wav" not found! (May be OK if not custom)
[WARNING] vrad.pack_file(): "bee2/sound/music/sp_a3_bomb_flings_z2c.wav" not found! (May be OK if not custom)
[WARNING] vrad.pack_file(): "bee2/sound/music/sp_a1_intro6.wav" not found! (May be OK if not custom)
[WARNING] vrad.pack_file(): "bee2/sound/music/sp_a3_bomb_flings_z2.wav" not found! (May be OK if not custom)
[WARNING] vrad.pack_file(): "bee2/sound/music/sp_a3_bomb_flings_z2b.wav" not found! (May be OK if not custom)
[INFO] vrad.pack_content(): Injecting "scripts/game_sounds_manifest.txt" into packfile.
[INFO] vrad.pack_content(): Injecting "scripts/BEE2_generated_music.txt" into packfile.
[INFO] vrad.pack_content(): Injecting "scripts/vscripts/bee2/auto_run.nut" into packfile.
[INFO] vrad.pack_content(): Injecting "scripts/vscripts/BEE2/cube_setmodel.nut" into packfile.
[INFO] vrad.pack_content(): Packing complete!
[INFO] vrad.mod_screenshots(): Using PeTI screenshot!
[INFO] vrad.mod_screenshots(): Making "..\portal2\puzzles\76561198274465226\untitled.jpg" replaceable...
[INFO] vrad.main(): Forcing Cheap Lighting!
[INFO] vrad.run_vrad(): Calling original VRAD...
[INFO] vrad.run_vrad(): "C:\Program Files (x86)\Steam\steamapps\common\Portal 2\bin\vrad_original.exe" -bounce 2 -noextra -hdr -game "C:/Program Files (x86)/Steam/steamapps/common/Portal 2/portal2" "C:\Program Files (x86)\Steam\steamapps\common\Portal 2\sdk_content\maps\preview.bsp"
[WARNING] vrad.run_vrad(): VRAD failed! (1)

**Comment by lcyLand** [comment]
in hammer it says the problem is vbsp, probably because it complains about no player start.

**Comment by LautaroL20** [comment]
Check out if the compiler files are being blocked by your antivirus, also try out the following:

1 - Open Beemod and click on: `File ----> Uninstall From Selected Game ----> "Are you sure you want to delete "PORTAL 2"? ----> YES`.

2 - Wait 1 minute while Beemod removes their resources from Portal 2.

3 - Verify the Game Cache Files. 

4 - Make sure to have downloaded and unzipped the latest Item's Packages and Application version of BEE2.4, and both must be from the same Pre-Release Version, otherwise bugs may occur. Links:

https://github.com/BEEmod/BEE2.4/releases (Application) 
https://github.com/BEEmod/BEE2-items/releases (Item's Packages)

Note that if you are running Beemod on Linux or Mac Os X, or even on 32 Bit versions of Windows, issues can occur. Beemod currently works well only on 64 Bit versions of Windows 7, 8 and 10.

5 - Click on: `File ----> Options ----> Development`, then uncheck the option: `Preserve Game Directories`.

6 - Now choose your favorite Items & Style and export your palette to the game.

7 - Make sure to save these steps in a text file in case you get similar issues in the future.

8 - Happy mapmaking!

**Comment by lcyLand** [comment]
I've already said it: clean installs won't do anything. TeamSpen has also confirmed that there is no need to re-install the mod now just to get it to work. Also, the problem is vrad. So I'm waiting to stop getting the same copy-pasted response.

**Comment by LautaroL20** [comment]
Check if your antivirus are blocking your compiler files, it is probably that the issue could be caused by your antivirus.

**Comment by lcyLand** [comment]
I've had the whole steam folder in my av exclusions for about 2 months now. So I don't think that's the problem.

**Comment by lcyLand** [comment]
It seems like every time I reinstalled it (not did a clean install but reinstall the program itself) something changed, now it wants to work. So I can close this now.

**Comment by lcyLand** [comment]
Well actually, it's having problems building again, here's my file for errors:

[compile errors.txt](https://github.com/BEEmod/BEE2.4/files/2268328/compile.errors.txt)

**Comment by TeamSpen210** [comment]
```
Too Many Overlays!
 | MAX_MAP_OVERLAYS = 512
```

Here's your problem. You have too many "overlays" in your map. Those are things like antlines, signage. Try rearranging items to simplify antlines (each straight section and each corner is a different antline), or remove items.

**Comment by lcyLand** [comment]
Ah. I wasn't paying attention to "overlays", and I know that antlines make up a lot of it. I'll change that.
